install.sh

```diff
@@ -1,63 +1,86 @@
+else
+  INDEX_IMAGE=${INDEX_IMAGE:-registry.redhat.io/redhat/community-operator-index:latest}
+  CATSRC_NAME=${CATSRC_NAME:-operators}
+  CATSRC_NAMESPACE="${CATSRC_NAMESPACE:-olm}"
 
-#5) Create a new image pull secret out of your local .docker/config.json file
-kubectl create secret generic community-operators-secrets -n olm --from-file=.dockerconfigjson=$HOME/.docker/config.json --type=kubernetes.io/dockerconfigjson
+  OPERATOR_NAMESPACE="${OPERATOR_NAMESPACE:-operators}"
+  CHANNEL="${CHANNEL:-beta}"
+  
+  #The SBO can also be installed in a vanilla kubernetes cluster. A prerequisite for this would be to add credentials for the registry.redhat.io in the cluster. Steps to be followed to achieve the same are listed below:
+  #1) Follow [Red Hat Container Registry Authentication Steps](https://access.redhat.com/RegistryAuthentication)
+  #2) Verify that your credentials are correct using docker login -u <your_username> -p <your_passwd> registry.redhat.io
+  docker login registry.redhat.io
 
-#6) Add that pull secret to the default account in olm namespace
-kubectl patch serviceaccount default -p '{"imagePullSecrets": [{"name": "community-operators-secrets"}]}' -n=olm
+  #3) Start a kubernetes cluster. Any Kubernetes cluster can be used as well. The script assumes that a Kubernetes is up and running and the user has logged into it.
+
+  #4) Enable [OLM](https://github.com/operator-framework/operator-lifecycle-manager) in the cluster by running the following command
+  curl -sL https://github.com/operator-framework/operator-lifecycle-manager/releases/download/0.16.1/install.sh | bash -s 0.16.1
+  #On minikube you can alternatively install OLM by running `minikube addons enable olm`
+
+  #5) Create a new image pull secret out of your local .docker/config.json file
+  kubectl create secret generic community-operators-secrets -n olm --from-file=.dockerconfigjson=$HOME/.docker/config.json --type=kubernetes.io/dockerconfigjson
+
+  #6) Add that pull secret to the default account in olm namespace
+  kubectl patch serviceaccount default -p '{"imagePullSecrets": [{"name": "community-operators-secrets"}]}' -n=olm
+fi
 
 #7) Install the operator by running the following commands
-#Apply CatalogSource for obtaining catalog of community operators
-kubectl apply -f - << EOD
+if [ -n "$INDEX_IMAGE" ]; then
+
+  # Apply CatalogSource
+  kubectl apply -f - << EOD
 ---
 apiVersion: operators.coreos.com/v1alpha1
 kind: CatalogSource
 metadata:
-  name: community-operators
-  namespace: olm
+  name: $CATSRC_NAME
+  namespace: $CATSRC_NAMESPACE
 spec:
-  displayName: Community Operators
-  image: registry.redhat.io/redhat/community-operator-index:latest
   sourceType: grpc
+  image: $INDEX_IMAGE
+  displayName: SBO Test
   publisher: Red Hat
-  updateStrategy:
-    registryPoll:
-      interval: 10m0s
 EOD
-#Apply subscription for subscribing to the beta channel of the service binding operator
+fi
+
+CSV_VERSION=$(./hack/get-current-csv.sh)
```

Comment:
we do not need this because we always want to install latest version available on the channel.

Reply:
In order to install the operator via subscription, CSV version is needed. This script looks into the package manifest for a given catalog source and it's channel and extract that version from it (that is the "latest version available on the channel" you mentined)... the above line is missing the arguments for catalog source and channel which I have fixed in the recent commit.

Reply:
Take it back... you are correct, we do not need CSV in subscription in our case.